spec/serializers/basic_reviewable_flagged_post_serializer_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+describe BasicReviewableFlaggedPostSerializer do
+  fab!(:topic) { Fabricate(:topic, title: "unsafe title <a>") }
+  fab!(:post) { Fabricate(:post, topic: topic) }
+
+  fab!(:reviewable) do
+    ReviewableFlaggedPost.needs_review!(target: post, topic: topic, created_by: Discourse.system_user)
+  end
+
+  def get_json
+    described_class.new(reviewable, root: false).as_json
+  end
+
+  it "includes post_number of the flagged post" do
+    expect(get_json[:post_number]).to eq(post.post_number)
+  end
+
+  it "includes topic_title of the flagged post" do
+    expect(get_json[:topic_title]).to eq("Unsafe title &lt;a&gt;")
+  end
+
+  it "is a subclass of BasicReviewableSerializer" do
+    expect(described_class).to be < BasicReviewableSerializer
+  end
+end
```

Comment:
```suggestion
  subject { described_class.new(reviewable, root: false).as_json }

  it "includes post_number of the flagged post" do
    expect(subject[:post_number]).to eq(post.post_number)
  end

  it "includes topic_title of the flagged post" do
    expect(subject[:topic_title]).to eq("Unsafe title &lt;a&gt;")
  end

  it "is a subclass of BasicReviewableSerializer" do
    expect(described_class).to be < BasicReviewableSerializer
  end
end
```

Reply:
Before I go too far and refactor the other spec files, can you check this https://github.com/discourse/discourse/pull/17630/commits/4ab075b56d29806b2bcf4ac3e548bbc9cb144a27?

Reply:
Yup that looks good :+1: